iot-hub/Quickstarts/read-d2c-messages/ReadDeviceToCloudMessages.js

```diff
@@ -1,55 +1,159 @@
+        const hostname = error.info && error.info.hostname;
+        if (!hostname) {
+          reject(error);
+        } else {
+          resolve(
+            `Endpoint=sb://${hostname}/;EntityPath=${iotHubName};SharedAccessKeyName=${SharedAccessKeyName};SharedAccessKey=${SharedAccessKey}`
+          );
+        }
+      } else {
+        reject(error);
+      }
+      connection.close().catch(() => {
+        /* ignore error */
+      });
+    });
+  });
+}
 
 var printError = function (err) {
   console.log(err.message);
 };
 
 // Display the message content - telemetry and properties.
 // - Telemetry is sent in the message body
-// - The device can add arbitrary application properties to the message
+// - The device can add arbitrary properties to the message
 // - IoT Hub adds system properties, such as Device Id, to the message.
 var printMessage = function (message) {
-  console.log('Telemetry received: ');
+  console.log("Telemetry received: ");
   console.log(JSON.stringify(message.body));
-  console.log('Application properties (set by device): ')
-  console.log(JSON.stringify(message.applicationProperties));
-  console.log('System properties (set by IoT Hub): ')
-  console.log(JSON.stringify(message.annotations));
-  console.log('');
+  console.log("Properties (set by device): ");
+  console.log(JSON.stringify(message.userProperties));
+  console.log("System properties (set by IoT Hub): ");
+  console.log(JSON.stringify(message.systemProperties));
+  console.log("");
 };
 
-// Connect to the partitions on the IoT Hub's Event Hubs-compatible endpoint.
-// This example only reads messages sent after this application started.
-var ehClient;
-EventHubClient.createFromIotHubConnectionString(connectionString).then(function (client) {
-  console.log("Successfully created the EventHub Client from iothub connection string.");
-  ehClient = client;
-  return ehClient.getPartitionIds();
-}).then(function (ids) {
-  console.log("The partition ids are: ", ids);
-  return ids.map(function (id) {
-    return ehClient.receive(id, printMessage, printError, { eventPosition: EventPosition.fromEnqueuedTime(Date.now()) });
+async function main() {
+  console.log("IoT Hub Quickstarts - Read device to cloud messages.");
+
+  const iotHubConnectionString = "{your Iot Hub connection string}";
+  const iotHubName = "{your Iot Hub name}";
+
+  // You can skip calling convertIotHubToEventHubsConnectionString() to do the conversion
+  // if you already have access to the Event Hubs compatible connection string from the
+  // Azure portal or the Azure CLI
+  // az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+  const eventHubsConnectionString = await convertIotHubToEventHubsConnectionString(
+    iotHubConnectionString
+  );
+
+  const consumerClient = new EventHubConsumerClient(
+    "Default",
+    eventHubsConnectionString,
+    iotHubName
+  );
+  consumerClient.subscribe({
+    processEvents: printMessage,
```

Comment:
Assuming `printMessage` is updated to `printMessages`
```suggestion
    processEvents: printMessages,
```